### ISSUE TITLE
Crash when trying to check malformed protocol link

### ISSUE THREAD

**Issue by nschonni** [tcort/markdown-link-check#46]
When the link checker hits a bad link, it crashes out

EX: A double H in `hhttps://docs.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2008-R2-and-2008/cc772207(v=ws.10)`
```console
C:\Users\Nicholas\AppData\Roaming\npm\node_modules\markdown-link-check\markdown-link-check:93
            if(result.status === 'dead') {
                      ^

TypeError: Cannot read property 'status' of null
    at C:\Users\Nicholas\AppData\Roaming\npm\node_modules\markdown-link-check\markdown-link-check:93:23
    at Array.forEach (<anonymous>)
    at C:\Users\Nicholas\AppData\Roaming\npm\node_modules\markdown-link-check\markdown-link-check:92:17
    at C:\Users\Nicholas\AppData\Roaming\npm\node_modules\markdown-link-check\node_modules\async\dist\async.js:1140:9
    at C:\Users\Nicholas\AppData\Roaming\npm\node_modules\markdown-link-check\node_modules\async\dist\async.js:473:16
    at iterateeCallback (C:\Users\Nicholas\AppData\Roaming\npm\node_modules\markdown-link-check\node_modules\async\dist\async.js:988:17)
    at C:\Users\Nicholas\AppData\Roaming\npm\node_modules\markdown-link-check\node_modules\async\dist\async.js:969:16
    at C:\Users\Nicholas\AppData\Roaming\npm\node_modules\markdown-link-check\node_modules\async\dist\async.js:1137:13
    at C:\Users\Nicholas\AppData\Roaming\npm\node_modules\markdown-link-check\index.js:65:13
    at linkCheck (C:\Users\Nicholas\AppData\Roaming\npm\node_modules\markdown-link-check\node_modules\link-check\index.js:24:9)
```
I've also noticed this for some custom protocol links too.

**Comment by tcort** [comment]
Ack. I see what's going on here. `link-check` is calling back with an error (unsupported protocol). `markdown-link-check` doesn't check `err` (yikes!) and blindly uses the results (eek!). I'll work on a fix this weekend if no one beats me to it. An unsupported protocol in a link should not be a fatal error (maybe a warning?), and other links should be processed.

**Comment by tcort** [comment]
Fix published in `v3.5.5`
